packages/ui-components/src/Input/Field.tsx

```diff
@@ -1,70 +1,38 @@
-import React, { FunctionComponent, useState } from "react";
+import React from "react";
 import { Field, FieldProps, FieldRenderProps } from "react-final-form";
-
-import {
-  SingleLineTextInput,
-  TextInputProps,
-  NumberInput,
-  NumberProps,
-} from "./TextTypeInput";
-import {
-  EmailInput,
-  NewPasswordInput,
-  ExistingPasswordInput,
-  EmailProps,
-  NewPasswordProps,
-  ExistingPasswordProps,
-  PasswordInputType,
-} from "./EmailPasswordInput";
-import { CheckboxInput, CheckboxInputProps } from "./CheckboxInput";
-import Icon from "../Icon/Icon";
-
-// the following omit statements can be removed
-// once type better limits what props can be passed in
-type TextInputFieldProps = Omit<
-  FieldProps<string, FieldRenderProps<string, HTMLElement>, HTMLElement> &
-    TextInputProps,
-  "type"
->;
-
-type NumberInputFieldProps = Omit<
-  FieldProps<number, FieldRenderProps<number, HTMLElement>, HTMLElement> &
-    NumberProps,
-  "type"
->;
-
-type CheckboxInputFieldProps = Omit<
-  FieldProps<boolean, FieldRenderProps<boolean, HTMLElement>, HTMLElement> &
-    CheckboxInputProps,
-  "type"
->;
-
-type EmailInputFieldProps = Omit<
-  FieldProps<string, FieldRenderProps<string, HTMLElement>, HTMLElement> &
-    EmailProps,
-  "type"
->;
-
-type NewPasswordInputFieldProps = Omit<
-  FieldProps<string, FieldRenderProps<string, HTMLElement>, HTMLElement> &
-    NewPasswordProps,
-  "type"
->;
-
-type ExistingPasswordInputFieldProps = Omit<
-  FieldProps<string, FieldRenderProps<string, HTMLElement>, HTMLElement> &
-    ExistingPasswordProps,
-  "type"
->;
-
-type InternalFieldProps =
-  | ({ InputFieldComponent: React.FunctionComponent } & TextInputFieldProps)
-  | CheckboxInputFieldProps
-  | EmailInputFieldProps
-  | NewPasswordInputFieldProps
-  | ExistingPasswordInputFieldProps;
-
-const InputField = ({
+import { SingleLineTextInput, NumberInput } from "./TextTypeInput";
+import { EmailInput, PasswordInput } from "./EmailPasswordInput";
+import { CheckboxInput } from "./CheckboxInput";
+
+// not including multiline input as it depends on Figma designs
+type InputTypes =
+  | typeof PasswordInput
+  | typeof EmailInput
+  | typeof SingleLineTextInput
+  | typeof NumberInput
+  | typeof CheckboxInput;
+
+// Extends field render props, which will be passed to the Field component in InputField
+export interface InputFieldProps<
+  IComponent extends React.ComponentType<InputTypes>
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+> extends FieldProps<any, FieldRenderProps<any, HTMLElement>, HTMLElement> {
```

Comment:
We have an `any` here because the type we should pass here is the `FieldValue`, which is `string | boolean | number`, for the different kind of Inputs we have. Not sure how to get rid of the any...